packages/cojson/src/SyncStateManager.ts

```diff
@@ -55,40 +52,17 @@ export class SyncStateManager {
     };
   }
 
-  getCurrentSyncState(peerId: PeerID, id: RawCoID) {
-    // Build a lazy sync state object to process the isUploaded info
-    // only when requested
-    const syncState = {} as SyncState;
-
-    const getIsUploaded = () =>
-      this.getIsCoValueFullyUploadedIntoPeer(peerId, id);
-
-    Object.defineProperties(syncState, {
-      uploaded: {
-        enumerable: true,
-        get: getIsUploaded,
-      },
-    });
-
-    return syncState;
-  }
-
-  triggerUpdate(peerId: PeerID, id: RawCoID) {
-    const peer = this.syncManager.peers[peerId];
-
-    if (!peer) {
-      return;
-    }
-
-    const peerListeners = this.listenersByPeers.get(peer.id);
+  triggerUpdate(peerId: PeerID, id: RawCoID, knownState: CoValueKnownState) {
+    const peerListeners = this.listenersByPeers.get(peerId);
 
     // If we don't have any active listeners do nothing
     if (!peerListeners?.size && !this.listeners.size) {
       return;
     }
 
-    const knownState = peer.knownStates.get(id) ?? emptyKnownState(id);
-    const syncState = this.getCurrentSyncState(peerId, id);
+    const syncState = {
+      uploaded: this.getIsCoValueFullyUploadedIntoPeer(knownState, id),
+    };
```

Comment:
Moved away from using the getter, as making the uploaded property lazy isn't necessary anymore.